src/api/textbook.js

```diff
@@ -18,7 +22,13 @@ export async function getTextbookById(id) {
   return null
 }
 
-export function getTextbooksByUserId(userId) {
-  // TODO implement
-  return []
-}
\ No newline at end of file
+export async function getTextbooksByUserId(userId) {
+  const textbookCollectionRef = collection(db, 'textbooks');
+  const q = query(textbookCollectionRef, or(where('seller_id', '==', userId), where('buyer_id', '==', userId)));
+  const textbooks = (await getDocs(q)).docs
+
+  return Promise.all(textbooks.map(textbook => processStatus(textbook.ref, textbook.data())))
+}
+
+// we should handle any race conditions that comes with textbook events
+// eg listing removal and reservation cancelled at the same time
```

Comment:
How do we plan to do this?

Reply:
seems like firstore has transactions that will allow us to do this: https://firebase.google.com/docs/firestore/manage-data/transactions

Main point:
> In the case of a concurrent edit, Cloud Firestore runs the entire transaction again. For example, if a transaction reads documents and another client modifies any of those documents, Cloud Firestore retries the transaction. This feature ensures that the transaction runs on up-to-date and consistent data.